### ISSUE TITLE
Bitwise operators implementations

### ISSUE THREAD

**Issue by ThakeeNathees** [ThakeeNathees/pocketlang#54]
## [TODO: bitwise operators](https://github.com/ThakeeNathees/pocketlang/blob/749456e215590218afa6dead42c1259007728ea3/src/pk_vm.c#L1249)

The bitwise operators (except for `&`) currently aren't implemented. Evaluating an expression like `a | b` will cause a crash with a `TODO: ` message. Which needs to be implemented.

Here is a simplified example of how `a | b` compiled into pocketlang [bytecode](https://en.wikipedia.org/wiki/Bytecode)
```
PUSH a  ; push value 'a' on the stack
PUSH b  ; push value 'b' on the stack
BIT_OR  ; pop the top 2 values on the stack, perform bitwise or and push the result.
```

And here is a simplified  "pseudo-code" of the pocketlang VM

```c
void runVM() {

  while (!finished)
    instruction = nextInstruction();
  
    switch (instruction)
      ...
      case PUSH:
        push(value()); // value a and b

      case BIT_AND:
        Var b = pop(); // from stack.
        Var a = pop(); // from stack.
        Var result = varBitAnd(a, b); // a & b;
        push(result);
        break;

      case BIT_OR:
      case BIT_XOR:
        TODO;
}
```

## How to implement

- Use `BIT_AND` ([here](https://github.com/ThakeeNathees/pocketlang/blob/749456e215590218afa6dead42c1259007728ea3/src/pk_vm.c#L1233-L1249)) as a reference to implement `BIT_OR`, `BIT_XOR`, ..
- Use `varBitAnd` ([here](https://github.com/ThakeeNathees/pocketlang/blob/749456e215590218afa6dead42c1259007728ea3/src/pk_core.c#L1089-L1101)) as a reference to implement `varBitOr`, `varBitXor`.

## Finally

Once you've successfully implemented, write some tests at [tests/lang/basics.pk](https://github.com/ThakeeNathees/pocketlang/blob/749456e215590218afa6dead42c1259007728ea3/tests/lang/basics.pk)
(note that currently, pocketlang doesn't support binary, hex literals)

```ruby
assert( 1 |  2 ==  3)
assert(10 | 12 == 14)
assert(-1 | 10 == -1)
```

**Comment by xSavitar** [comment]
@ThakeeNathees, once I finish implementing all the bitwise operators, we can close this. I'll be implementing them one at a time and tagging you for review. Above is the first one (bitwise OR) implemented.

**Comment by ThakeeNathees** [comment]
Whoops, closed by the ["Closing Issues via Commit Messages"](https://github.blog/2013-01-22-closing-issues-via-commit-messages/) Feature

**Comment by ThakeeNathees** [comment]
This is fully implemented at #59 and #63.
